compiler/runtime.go

```diff
@@ -101,7 +105,9 @@ func (e *blockEvaluator) Eval(ctx context.Context, scope Scope) (values.Value, e
 		if err != nil {
 			return nil, err
 		}
+		value.Release()
 	}
+	value.Retain()
```

Comment:
I think this retain is wrong. Eval should return a value with a reference count of 1. We're the only one with a reference to it. Since we pass our reference back to the caller, we would both "retain" it to hand it to outside and "release" it internally which would just be a move operation and would not involve a retain.

The release above this looks correct. Although it looks strange. I do think it's correct.

Reply:
This confuses me. Each instance of `value` starts with a reference count of 1, and we release it at the end of the loop, so each value would have a reference count of 0 by the time the loop is done.

...but because we're returning the last instance of `value`, I would think we'd want to retain it to bring its reference count back up to 1. Am I missing something?

Maybe I should change the code to release all but the last instance of `value`?

Reply:
This would be correct. Also, I didn't realize this the first time, but this order of operations is incorrect. You can't release a value and then retain it as the reference count would have potentially dropped to zero.

You might just want to reorder the loop so it's not assigning values that it will ultimately discard to the variable.